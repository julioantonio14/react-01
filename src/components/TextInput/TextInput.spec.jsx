import { fireEvent, render, screen } from "@testing-library/react";
import { TextInput } from ".";
import userEvent from "@testing-library/user-event";

describe("<TextInput />", () => {
    it("should have a value of searchValue", () => {
        const fn = jest.fn();
        render(<TextInput searchValue={"Testando"} handleChange={fn} />);
        const input = screen.getByPlaceholderText(/Escreva Alguma coisa/i);
        expect(input).toBeInTheDocument();
        expect(input.value).toBe("Testando");
    });
    it("should call handleChange On each key pressed", () => {
        const fn = jest.fn();
        render(<TextInput handleChange={fn} />);
        const input = screen.getByPlaceholderText(/Escreva Alguma coisa/i);
        const value = "o valor";
        userEvent.type(input, value);
        expect(input.value).toBe(value);
        expect(fn).toHaveBeenCalledTimes(value.length)
    });
    it("should match snapshot", () => {
        const fn = jest.fn();
        const {container} = render(<TextInput handleChange={fn} />);
        expect(container).toMatchSnapshot();
    });
});
