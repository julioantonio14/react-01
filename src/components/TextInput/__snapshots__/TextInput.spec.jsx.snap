// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> should match snapshot 1`] = `
<div>
  <input
    class="text-input"
    id=""
    name=""
    placeholder="Escreva Alguma coisa"
    type="search"
    value=""
  />
</div>
`;
